### ISSUE TITLE
Find the nearest neighbor and SVD to solve for the transformation

### ISSUE THREAD

**Issue by BiaoBiaoLi** [yewzijian/RPMNet#8]
Hi yewzijian:
Thanks your work,I want ask  two questions:

1. As for finding the nearest neighbor:
    group_idx[sqrdists > radius ** 2] = N
    group_idx = group_idx.sort(dim=-1)[0][:, :, :nsample]
In the process, did you not use the 64 closest points? just took the 64  points that satisfy the distance requirement?

2.SVD to solve for the transformation:
In this function
transform = compute_rigid_transform(xyz_src, weighted_ref, weights=torch.sum(perm_matrix, dim=2))
cov = a_centered.transpose(-2, -1) @ (b_centered * weights_normalized)
the cov should be cov = a_centered.transpose(-2, -1) @ b_centered ?

**Comment by yewzijian** [comment]
1. Yes, we select up to 64 points that within the desired radius as in the original PointNet++. This works better than using the closest points in practice.
2. The covariance is weighted so we scale the contribution of each correspondence by its weight.

**Comment by BiaoBiaoLi** [comment]
ok,thanks

**Comment by BiaoBiaoLi** [comment]
Thanks your answer,I also want to konw the the if the codes of RPM in your paper is open source.

**Comment by yewzijian** [comment]
I think you are referring to the license? The source code uses a MIT license, as seen in the license file in the repository.

**Comment by BiaoBiaoLi** [comment]
Sorry,I don't  make myself clear. I want to  konw  if the codes of RPM(New algorithms for 2D and 3D
point matching: pose estimation and correspondence) in your paper is open source. I can't find it in Internet.

**Comment by yewzijian** [comment]
Oh, you mean the codes of the original RPM? The author's codes can be found [here](https://www.cise.ufl.edu/~anand/students/chui/tps-rpm.html). This version is written for non-rigid point cloud registration.

For the results in my paper, I used my own implementation.

**Comment by BiaoBiaoLi** [comment]
Yes,thank you so much, I find it for a long time.

**Comment by BiaoBiaoLi** [comment]
Hi yewzijian:
Thanks your work,I also want to know why sinkhorn use  logarithms to normalize the matrices?
what i read the original RPM is showed below:
![image](https://user-images.githubusercontent.com/44672683/94993438-8d297c80-05c3-11eb-8d7b-fa6c5fcfe96d.png)
Can you tell me what is the difference?

**Comment by yewzijian** [comment]
We perform the normalization in log space in the actual implementation for numerical stability. The operations are otherwise the same, e.g. the division in linear space is changed to subtraction in log space.

Zi Jian
